subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/classpath/SplitClasspathEntrySnapshotCache.java

```diff
@@ -49,11 +49,25 @@ public ClasspathEntrySnapshot get(File file, HashCode hash) {
     }
 
     @Override
-    public ClasspathEntrySnapshot get(File entry, Factory<ClasspathEntrySnapshot> factory) {
-        if (globalCacheLocations.isInsideGlobalCache(entry.getPath())) {
-            return globalCache.get(entry, factory);
+    public ClasspathEntrySnapshot get(File entry, Function<? super File, ? extends ClasspathEntrySnapshot> factory) {
+        return getCacheFor(entry).get(entry, factory);
+    }
+
+    @Override
+    public ClasspathEntrySnapshot get(File key) {
+        return getCacheFor(key).get(key);
+    }
```

Comment:
We could have a default implementation for this guy in `Cache.get()`.

Reply:
What would be the default implementation?